src/ifvc.c

```diff
@@ -268,6 +270,20 @@ void buildIfVc(void) {
             mask = ((struct sockaddr_in *)&IfReq.ifr_addr)->sin_addr.s_addr;
             subnet = addr & mask;
 
+            dp = getIfByName(IfPt->ifr_name, 1);
+            if (dp != NULL && dp->allowednets != NULL) {
+                allowednet = (struct SubnetList *)malloc(sizeof(struct SubnetList));
+                if (allowednet == NULL) my_log(LOG_ERR, 0, "Out of memory !");
```

Comment:
Maybe you could explain the problem in the above code to me, because my C is quite rusty and I clearly fail to see it.